code/modules/hallucinations/effects/major.dm

```diff
@@ -408,3 +408,125 @@
 	if(images && images[1])
 		images[1].icon = 'icons/mob/alien.dmi'
 		images[1].icon_state = "alienh_pounce"
+
+/obj/effect/hallucination/blob
+	duration = 30 SECONDS
+	/// The blob zombie hallucination.
+	var/obj/effect/hallucination/chaser/attacker/blob_zombie/zombie = null
+	/// The self delusion blob zombie hallucination, triggers on knockdown.
+	var/obj/effect/hallucination/blob_zombify/player_zombie = null
+	/// List of turfs that need expanding from.
+	var/list/turf/expand_queue = list()
+	/// Associative list of turfs that have already been processed.
+	var/list/turf/processed = list()
+	/// The delay at which the blob expands in deciseconds. Shouldn't be too low to prevent lag.
+	var/expand_delay = 7.5 SECONDS // Expand 6 times
+	/// Expand timer handle.
+	var/expand_timer = null
+
+/obj/effect/hallucination/blob/Initialize(mapload, mob/living/carbon/target)
+	. = ..()
+	var/list/locs = list()
+	for(var/turf/T in oview(world.view / 2, target))
+		var/light_amount = min(1, T.get_lumcount()) - 0.5
+		if(!is_blocked_turf(T) && light_amount <= 0)
+			locs += T
+	if(!length(locs))
+		qdel(src)
+		return
+	var/turf/T = get_turf(pick(locs))
+	var/turf/Z = get_turf(pick(locs))
+	color = pick(	COLOR_BLACK, COLOR_RIPPING_TENDRILS, COLOR_BOILING_OIL, COLOR_ENVENOMED_FILAMENTS, COLOR_LEXORIN_JELLY,
+								COLOR_KINETIC_GELATIN, COLOR_CRYOGENIC_LIQUID, COLOR_SORIUM, COLOR_TESLIUM_PASTE )
+	create_blob(T)
+	create_zombie(Z)
+	expand_queue += T
+	processed[T] = TRUE
+	expand_timer = addtimer(CALLBACK(src, PROC_REF(expand)), expand_delay, TIMER_LOOP | TIMER_STOPPABLE)
+
+/obj/effect/hallucination/blob/Destroy()
+	deltimer(expand_timer)
+	QDEL_NULL(expand_queue)
+	QDEL_NULL(processed)
+	QDEL_NULL(zombie)
+	QDEL_NULL(player_zombie)
+	return ..()
+
+/**
+  * Called regularly in a timer to process the blob expanding.
+  */
+/obj/effect/hallucination/blob/proc/expand()
+	// Brace for potentially expensive proc
+	for(var/t in expand_queue)
+		var/turf/source_turf = t
+		expand_queue -= source_turf
+		//Expand to each dir
+		for(var/direction in GLOB.cardinal)
+			var/turf/target_turf = get_step(source_turf, direction)
+			if(processed[target_turf] || !source_turf.CanAtmosPass(direction) || !target_turf.CanAtmosPass(turn(direction, 180)))
+				continue
+			create_blob(target_turf)
+			expand_queue += target_turf
+			processed[target_turf] = TRUE
+
+/**
+  * Creates a fake blob overlay on the given turf.
+  *
+  * Arguments:
+  * * T - The turf to create a fake plasma overlay on.
+  */
+/obj/effect/hallucination/blob/proc/create_blob(turf/T)
+	var/image/I = image('icons/mob/blob.dmi', T, "blob", layer = FLY_LAYER)
```

Comment:
Might want to make it play the vomit sound just to really sell it

Reply:
Are such modifications allowed after the PTBF deadline?

Reply:
Yep!

Reply:
Yeah, we just wanted the (mostly) formed ideas to be out in a PR before the deadline. Changes that don't really alter the fundamental idea of the PR are fine!

Reply:
I've added a vomit sound effect on the initial spawn, as requested.